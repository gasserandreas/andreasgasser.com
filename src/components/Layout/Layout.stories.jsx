import React from 'react';

import Layout from './Layout';
import LongText from './__examples__/LongText';
import Page from '../Page/Page';

export default {
  title: 'Layout',
  component: Layout,
  decorators: [(storyFn) => (
    <div
      style={{
        height: '110vh',
      }}
    >
      <Page>{storyFn()}</Page>
    </div>
  )],
};

export const defaultLayout = () => (
  <Layout>
    <p>default Layout</p>
  </Layout>
);

export const longContent = () => (
  <Layout>
    <LongText />
  </Layout>
);

defaultLayout.story = {
  name: 'default',
};

// export const custom = () => (
//   <Layout>
//     <p>custom Layout</p>
//   </Layout>
// );
