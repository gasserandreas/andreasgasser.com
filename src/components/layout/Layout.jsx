import React from 'react';
import PropTypes from 'prop-types';
import { useStaticQuery, graphql } from 'gatsby';
import styled, { ThemeProvider } from 'styled-components';
import { Box } from 'rebass/styled-components';

import Header from '../header/Header';
import Footer from '../footer/Footer';
import GlobalStyle from '../GlobalStyles';
import theme from '../Theme/theme';

const Styled = {
  Wrapper: styled(Box)`
    height: 100%;
    min-width: 100%;
    text-align: center;
  `,
  Content: styled(Box)`
    margin: 0 auto;

    main {
      margin: 2rem 0;
    }
  `,
};

// simple layout component
export const Layout = ({ children, data }) => (
  <ThemeProvider theme={theme}>
    <GlobalStyle />
    <Styled.Wrapper>
      {/* Header */}
      <Header siteTitle={data.site.siteMetadata.title} />
      {/* Main content */}
      <Styled.Content>
        <main>{children}</main>
      </Styled.Content>
      <Footer />
    </Styled.Wrapper>
  </ThemeProvider>
);

Layout.propTypes = {
  children: PropTypes.node.isRequired,
  data: PropTypes.shape({
    site: PropTypes.shape({
      siteMetadata: PropTypes.shape({
        title: PropTypes.string,
      }),
    }),
  }),
};

Layout.defaultProps = {
  data: {
    site: {
      siteMetadata: {
        title: '',
      },
    },
  },
};

// layout with static data
const ConnectedLayout = ({ children }) => {
  const data = useStaticQuery(graphql`
    query SiteTitleQuery {
      site {
        siteMetadata {
          title
        }
      }
    }
  `);

  return (
    <Layout staticData={data}>{children}</Layout>
  );
};

ConnectedLayout.propTypes = {
  children: PropTypes.node.isRequired,
};

export default ConnectedLayout;
