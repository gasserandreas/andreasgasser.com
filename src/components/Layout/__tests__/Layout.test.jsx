/* global withTheme */
import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Layout from '../Layout';

const findById = (dom, testid) => dom.find(`[data-testid="${testid}"]`);

describe('components/Layout', () => {
  it('render responsive layout', () => {
    const dom = mount(withTheme(<Layout />));
    expect(dom.exists()).toBeTruthy();

    const leftNavigation = findById(dom, 'leftNavigation').at(0);
    expect(toJson(leftNavigation.html())).toMatchSnapshot();
  });

  // it('simple snapshot test', () => {
  //   const dom = mount(<Footer />);
  //   expect(toJson(dom)).toMatchSnapshot();
  // });
});
