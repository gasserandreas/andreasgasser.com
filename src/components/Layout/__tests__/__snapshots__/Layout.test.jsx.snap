// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Layout render responsive layout 1`] = `undefined`;
